host/CMakeLists.txt

```diff
@@ -452,12 +452,12 @@ add_library(oehostapp INTERFACE)
 target_link_libraries(oehostapp INTERFACE oehost)
 
 if (UNIX)
-  target_link_libraries(oehostapp INTERFACE
-      -rdynamic
+  target_link_libraries(oehost INTERFACE
       -Wl,-z,noexecstack)
+  target_link_libraries(oehostapp INTERFACE
+      -rdynamic)
   target_link_libraries(oehostverify INTERFACE
-      -rdynamic
-      -Wl,-z,noexecstack)
+      -rdynamic)
```

Comment:
`oehostverify` parallels `oehost` and does not link `oehost`, so it still needs its own copy of the `-Wl,-z,noexecstack` arguments.

Tangent: @anakrish I'm not sure why we decided to create a target library just for the purpose of tacking on the `-rdynamic` linker option. It seems like a weird indirection and it seems like an odd choice that we default all our samples to use it?

Reply:
Ah you're right, added the flag back to oehostverify.

Agreed that I don't see a reason for having both targets. If there isn't a good reason I can open an issue/pr to remove it (although it is a breaking change so maybe easier said than done). But I'll wait for Anand's response.

Reply:
@CodeMonkeyLeet  
oehostapp was done by jorkl 2 years ago https://github.com/openenclave/openenclave/commit/5f0cded986953dc765bf166b1f4f77c5d646ccf0

I am not sure why we would need the separation between oehost and oehostapp.

However, we do need the separation between oehostverify and oehostapp. The former allows attestation by a host that is not running on sgx hardware. The latter allows creating enclave applications.

Reply:
@anakrish let me refine the question: we currently expect everyone to link against oehostapp given the sample usage anyway, and so we ostensibly already recommend `-rdynamic` to be the default use case. We still make oehost available, presumably to provide an option without `-rdynamic` (and prior to this change, without `noexecstack`). 
- Do you think providing a library that doesn't pull in `-rdynamic` in CMake by default is valuable?
    - My understanding of `-rdynamic` is that it pulls in all symbols which is useful for backtracing and dynamic library load/link scenarios and is the preferable default.
- I assume this expands the size of the resulting binary; are there other potentially undesirable effects?
    - Assuming we only have the `-rdynamic` default, is there an easy way to override that in a user's build? (e.g. I couldn't find a `no-dynamic` option).

@jhand2 Yes, it would be a breaking change to remove one or the other library, but it's one of those things that's better to change sooner rather than later, and we would still want to mark deprecation in this release, then eliminate in the next one.

Reply:
I opened #2595 to track. This issue aside, are we ok with this PR going in?

Reply:
@jhand2 Thanks for creating the issue #2595 . I have added the _first good issue_ label to it.
Though it is not readily obvious why SGX driver (which deals with creating enclaves) should be impacted by execstack bits of the host executable, it is good for this change to go in since we had intended noexecstack to be applied to hosts and we weren't ensuring it.

Reply:
@CodeMonkeyLeet I don't have answer to the significance of -rdynamic. Yes, it seems relevant only if a dynamically loaded (via dlopen) shared library were to depend on symbols in the main host executable. But I am not sure if it is applicable in some way specifically for enclave applications. Maybe the global enclave list needs to be visible to all modules even though it exists only in the main executable?
Not really sure. Hopefully, we can investigate it as part of #2595